### PR TITLE
Switch to logging v2

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Switch to logging v2

### CODE REVIEW

File: c_pal_ll/interfaces/reals/CMakeLists.txt [hunk 1]
```diff
@@ -10,4 +10,4 @@ set(reals_pal_ll_h_files
 
 add_library(pal_ll_interfaces_reals INTERFACE)
 target_include_directories(pal_ll_interfaces_reals INTERFACE ${CMAKE_CURRENT_LIST_DIR})
-target_link_libraries(pal_ll_interfaces_reals INTERFACE c_logging)
+target_link_libraries(pal_ll_interfaces_reals INTERFACE c_logging INTERFACE c_logging_v2)
```
Comment (darobs) [inline c_pal_ll/interfaces/reals/CMakeLists.txt:13]: >INTERFACE c_logging [](http://example.com/codeflow?start=46&length=19)

Question: is c_logging needed here?

**Review by darobs — APPROVED** [review]
:shipit:

File: c_pal_ll/interfaces/reals/CMakeLists.txt (hunk 1 above)
Comment (dcristoloveanu) [reply c_pal_ll/interfaces/reals/CMakeLists.txt]: Should not be needed, removed

File: build_functions/c-pal_suppressions.sup [hunk 2]
```diff
@@ -4,6 +4,15 @@
    fun:thread_worker_func
    ...
 }
+{
+   <vfprintf suppression>
+   Helgrind:Race
```
[the remark this answers is not in the document]
Comment (dcristoloveanu) [reply build_functions/c-pal_suppressions.sup:9]: This is for suppressing Helgrind errors in printf. https://valgrind.org/docs/manual/hg-manual.html

File: common/tests/refcount_ut/some_refcount_impl.c [hunk 3]
```diff
@@ -2,9 +2,13 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 #include <stddef.h>
+
+#include "macro_utils/macro_utils.h"
```
[the remark this answers is not in the document]
Comment (dcristoloveanu) [reply common/tests/refcount_ut/some_refcount_impl.c:6]: IWYU wants it. And I am not going to fight it.

File: linux/tests/threadpool_linux_int/threadpool_linux_int.c [hunk 4]
```diff
@@ -85,9 +88,8 @@ static void threadpool_task_wait_random(void* parameter)
 
 static void work_function(void* context)
 {
-    volatile_atomic int32_t* call_count = (volatile_atomic int32_t*)context;
-    (void)interlocked_increment(call_count);
-    wake_by_address_single(call_count);
+    (void)interlocked_increment(&g_call_count);
```
[the remark this answers is not in the document]
Comment (dcristoloveanu) [reply linux/tests/threadpool_linux_int/threadpool_linux_int.c:91]: Hmmm, interesting that the compiler did not complain. I added -Wunused-parameter and the compiler seems to start flagging these. We likely have to add it to c-build-tools. Here is the task for it:

[24404485](https://msazure.visualstudio.com/One/_workitems/edit/24404485)	Task	Add add_compile_options(-Wunused-parameter) to c-build-tools	To Do	 	 	 	 	 	One

File: linux/src/sysinfo_linux.c [hunk 5]
```diff
@@ -3,7 +3,9 @@
 #include <inttypes.h>
 #include <unistd.h>
 
-#include "c_logging/xlogging.h"
+#include "macro_utils/macro_utils.h"
```
[the remark this answers is not in the document]
Comment (dcristoloveanu) [reply linux/src/sysinfo_linux.c:6]: Apparently it is, IWYU is complaining about it.

**Pull request merged**